pkg/cmd/pr/merge/merge.go

```diff
@@ -253,37 +269,50 @@ func (m *mergeContext) merge() error {
 		setCommitBody: m.opts.BodySet,
 	}
 
+	if m.addToMergeQueue() && !m.opts.InteractiveMode {
+		_ = m.warnf("%s A merge strategy cannot be set when adding a pull request to the merge queue\n", m.cs.FailureIconWithColor(m.cs.Red))
+		return cmdutil.SilentError
+	}
+
 	// get user input if not already given
 	if m.opts.InteractiveMode {
-		apiClient := api.NewClientFromHTTP(m.httpClient)
-		r, err := api.GitHubRepo(apiClient, m.baseRepo)
-		if err != nil {
-			return err
-		}
-
-		payload.method, err = mergeMethodSurvey(r)
-		if err != nil {
-			return err
-		}
-
-		m.deleteBranch, err = deleteBranchSurvey(m.opts, m.crossRepoPR, m.localBranchExists)
-		if err != nil {
-			return err
-		}
+		if m.addToMergeQueue() {
+			// auto merge will either enable auto merge or add to the merge queue
+			payload.auto = true
```

Comment:
Was this intentionally nested under `if InteractiveMode`? For simplicity and clarity, could the parent conditional have been:
```go
if m.addToMergeQueue() {
  payload.auto = true
} else if m.opts.InteractiveMode {
  // old code
}
```